test/unit/Cardano/WalletSpec.hs

```diff
@@ -188,5 +209,6 @@ instance Arbitrary AddressPoolGap where
 
 instance Arbitrary WalletId where
```

Comment:
why don't we share `instance Arbitrary WalletId` between specs ? (ie, the same is defined in TypeSpec)

Reply:
We don't share instances between tests to help keeping them separate and well-scoped. Every test file is sufficient to itself and doesn't need extra context. 
Having shared generator is usually a bad idea because, depending on the properties you're testing, you don't want the generators to be identical (we do actually have, in several places, generators for same data types that are completely different). Even in this PR, the generator for wallet Id in the MVarSpec generates much more conflicting wallet ids than in others. This is on purpose.

Generally speaking, we don't want to over-engineer our test suite; every tests are already complex enough to also share the burden of code architecture and structuring. Ideally, we would use generators as data and not class (like Hedgehog does it), but we are quite acquainted to QC (and the tooling seems also of better quality here..) so, we went for having orphan instances NOT shared between tests modules.